gneiss/tests/test_model.py

```diff
@@ -0,0 +1,175 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, gneiss development team.
+#
+# Distributed under the terms of the GPLv3 License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+
+import pandas as pd
+import statsmodels.formula.api as smf
+from skbio import TreeNode
+from gneiss._model import Model
+import unittest
+import pandas.util.testing as pdt
+import numpy.testing as npt
+
+
+# create some mock classes for testing
+class submock_ok(Model):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def summary(self):
```

Comment:
The purpose of this function is to force all of the subclasses to implement a summary method.

In the context of regression, we'll want to print out some information about good-of-fit (i.e. r^2),  number of balances, number of coefficients, ...

However, all of this functionality will be implemented in the subclasses - which will come later.